main/test/ca/uwaterloo/flix/language/phase/TestRedundancy.scala

```diff
@@ -1767,6 +1767,98 @@ class TestRedundancy extends FunSuite with TestUtils {
     expectError[RedundancyError.RedundantCheckedTypeCast](result)
   }
 
+  test("RedundantCheckedEffectCast.01") {
+    val input =
+      """
+        |def f(): Unit =
+        |    let _ =
+        |        if (true)
+        |            checked_ecast(x -> x)
+        |        else
+        |            x -> x;
```

Comment:
I don't understand how all the redundant checked effect casts succeed when they should fail (i.e. they compile, so the test fails). A lambda expression is pure and has the empty set as effect, thus effect casting a similar branch should have the same purity and effect, triggering the redundancy error. But that doesn't happen for some reason. Is the purity and effect set a type variable when effect casting?

See https://github.com/flix/flix/blob/a9b17efdb3a9cb9b810d82361023f8ba89abc942/main/src/ca/uwaterloo/flix/language/phase/Redundancy.scala#L648 for which rule in Redundancy, I'm referring to.

Reply:
I have not looked at all the code, but ecasting a lambda looks suspicious. You should probably only be ecasting the body.

Reply:
That's on purpose specifically to test it

Reply:
I looked at it, but I don't get it :'(

Reply:
I see. Well good thing we tested it

Reply:
If-then-else does not require both branches to have the same effect. So we have this inference:

* branch 1 `x -> x` is inferred pure,
* `checked_ecast(x -> x)` is inferred some free effect variable `e`
* branch 2 `x -> x` is inferred pure
* the whole if-then-else is inferred `e and pure` which is just `e`

But note that branch 1 `x -> x` and `checked_ecast(x -> x)` are inferred to have different effects, which is what causes the issue here

Reply:
Right. So. Uhm. What should we do?

Reply:
The test as it stands is invalid. Once the issue with region levels is fixed, it would be valid. For now we have to find a different way to test.

Reply:
Lets mark the tests as `ignored` then